### ISSUE TITLE
Prevent naming variables w/ existing native names

### ISSUE THREAD

**Issue by pdehaan** [eslint/eslint#10265]
**Tell us about your environment**
Node.js

* **ESLint Version:** <kbd>$ eslint --version #  v4.19.1</kbd>
* **Node Version:** <kbd>$ node --version # v8.11.1</kbd>
* **npm Version:** <kbd>$ npm --version # 5.6.0</kbd>

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  env: {
    browser: true,
    es6: true,
    node: true,
  },
  extends: [
    'eslint:recommended',
  ],
  root: true,
  rules: {
    'no-global-assign': 'error',
    'no-native-reassign': 'error',
    'no-redeclare': ['error', {builtinGlobals: true}],
    'no-shadow': 'error',
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/* eslint-disable no-console */

const Error = "uh oh";

console.log(Error);

try {
  throw new Error("what the what??");
} catch (err) {
  console.error(err.message);
}
```

```bash
$ npm test

> eslint-error-override@1.0.0 test /Users/pdehaan/dev/tmp/del/eslint-error-override
> eslint .

```

**What did you expect to happen?**

I was hoping ESLint would complain about me foolishly naming a variable `Error`, which is generally a silly variable name.

**What actually happened? Please include the actual, raw output from ESLint.**
Nothing. No errors.

```sh
$ npm test

> eslint-error-override@1.0.0 test /Users/pdehaan/dev/tmp/del/eslint-error-override
> eslint .

$ echo $? # 0

$ node index
uh oh
Error is not a constructor
```

**Comment by platinumazure** [comment]
Hi @pdehaan, thanks for the issue.

In your case, you are shadowing an existing global, not reassigning. Try enabling the [`builtinGlobals` option](https://eslint.org/docs/rules/no-shadow#builtinglobals) of the `no-shadow` rule; that should get you what you need.

**Comment by pdehaan** [comment]
@platinumazure: That seems to fix it, thanks!
I had set the `{builtinGlobals: true}` option on the **no-redeclare** rule, but completely missed it as an option on **no-shadow**.
